ROMFS/CMakeLists.txt

```diff
@@ -179,17 +184,35 @@ foreach(board_rc_file ${OPTIONAL_BOARD_RC})
 
 endforeach()
 
+list(APPEND extras_dependencies
+	${config_romfs_extra_dependencies}
+	)
+
+if (config_romfs_extra_files)
+	set(extras_copy_cmd COMMAND ${CMAKE_COMMAND} -E copy_if_different ${config_romfs_extra_files} ${romfs_gen_root_dir}/extras/)
+else()
+	set(extras_copy_cmd "")
+endif()
 add_custom_command(OUTPUT romfs_extras.stamp
 	COMMAND ${CMAKE_COMMAND} -E make_directory ${romfs_gen_root_dir}/extras/
 	COMMAND ${CMAKE_COMMAND} -E make_directory ${PX4_BINARY_DIR}/romfs_extras/
 	COMMAND ${CMAKE_COMMAND} -E copy_directory ${PX4_BINARY_DIR}/romfs_extras/ ${romfs_gen_root_dir}/extras/
+	${extras_copy_cmd}
```

Comment:
TIL. CMake continues to surprise me (not in a good way).

Side note, the dependencies here were never quite right. I believe we'll ultimately need to have cmake more directly handle the pruning to do it perfectly.